code/modules/research/xenobiology/xenobiology.dm

```diff
@@ -546,26 +547,31 @@
 	while(loc)
 		if(duration)
 			for(var/mob/living/M in orange (freezerange, src.loc))
-				if(M in immune)
-					continue
-				M.stunned = 10
-				M.anchored = 1
-				if(istype(M, /mob/living/simple_animal/hostile))
-					var/mob/living/simple_animal/hostile/H = M
-					H.AIStatus = AI_OFF
-					H.LoseTarget()
+				if(!istype(M, /mob/living/simple_animal/hostile/guardian/punch))
```

Comment:
no other way, because the timestop effect ends hostile mob AI, under which the guardian falls. Even giving them timestop is not enough to circumvent this.